src/pages/discussion/discussion.js

```diff
@@ -0,0 +1,263 @@
+import React, { Component } from "react";
+import "./discussion.scss";
+import Sidebar from "../../components/sidebar/sidebar";
+import { Container, Row, Col, Badge } from "react-bootstrap";
+import Avatar from "@material-ui/core/Avatar";
+import EditorChat from "../../components/editor/editor";
+import socketIOClient from "socket.io-client";
+import { IoMdArrowRoundBack } from "react-icons/io";
+import { FaCommentAlt, FaReply } from "react-icons/fa";
+import ReactHtmlParser, {
+  processNodes,
+  convertNodeToElement,
+  htmlparser2,
+} from "react-html-parser";
+
+var socket;
+
+function UserInfos(chats) {
+  return [
+    ...new Set(
+      chats.map((chat) => {
+        return chat.username;
+      })
+    ),
+  ];
+}
+
+class Discussion extends Component {
+  constructor(props) {
+    console.log(props);
+    super(props);
+    this.state = {
+      endpoint: "http://localhost:8000/",
```

Comment:
Please keep your state minimal. In your state only keep stuff which changes and you would want your component to re-render when it changes. There are many other places you can keep this stuff, maybe as a default prop, a class property or a global constant but not in state, please.